hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/DeleteHoodieTableCommand.scala

```diff
@@ -36,9 +37,13 @@ case class DeleteHoodieTableCommand(deleteTable: DeleteFromTable) extends Hoodie
 
     // Remove meta fields from the data frame
     var df = removeMetaFields(Dataset.ofRows(sparkSession, table))
-    if (deleteTable.condition.isDefined) {
-      df = df.filter(Column(deleteTable.condition.get))
+    // SPARK-38626 DeleteFromTable.condition is changed from Option[Expression] to Expression in Spark 3.3
+    val condition: Expression = deleteTable.condition match {
+      case option: Option[Expression] => option.getOrElse(null)
+      case expr: Expression => expr
+      case _ => throw new IllegalArgumentException(s"DeleteFromTable.condition has to be either Option[Expression] or Expression")
```

Comment:
Is this because of the API compatibility, so both `Option[Expression]` and `Expression` typed value can exist?  Nevertheless, looks not clean.

Reply:
Yes, in spark 3.3 this was changed from `Optional[Expression]` to `Expression`

Reply:
Added a new method `extractCondition` in `SparkAdapter` to clean this part